proposal.md

```diff
@@ -0,0 +1,283 @@
+# Reflect Library Improvements Proposal
+
+## Background
+
+In Eventide's libraries, Reflect has 6 use cases, but 4 of them do not utilize it in the designed way. Instead of making a use of the reflection's actuator, the constant is pulled from the reflection.
+
+The actuator is designed to pass the subject into the method being invoked. Subject could be overriden when needed.
+
+```ruby
+class SomeThing
+  module Validate
+    def self.call(some_thing)
+      # validate some_thing
+    end
+  end
+end
+
+some_thing = SomeThing.new
+
+validator_reflection = Reflect.(some_thing, :Validate)
+
+validator_reflection.(:call)
+```
+
+However, the actuator doesn't support methods with zero arity or methods with arity greater than one. Due to these limitations, the constant is extracted from the reflection, bypassing the actuator.
+
+It has been observed that the reflection's actuator should support methods with any arity and allow the subject to be overridden when needed.
```

Comment:
Consider: If the reflection's actuator were to support invoking protocol methods with any arity, then most use cases of the reflect library would be able to use the actuator and avoid disencapsulating the reflection's target constant.